mkdocs.yml

```diff
@@ -249,6 +246,8 @@ markdown_extensions:
   - tables
   # - mdx_truly_sane_lists
 extra:
+  version:
+    provider: mike
```

Comment:
@lenny-intel and @ernestojeda - I spent some time researching and playing with this over the weekend.  The old {{version}} doesn't work in mike to reference the latest version in a repo like it did the previous versioning system in mkdocs.  By default, it tries to defer to mike but mike plugin doesn't really exist - replacing {{version}} with {'provider':'mike'}.  I tried to use the version default (or any other label in the mkdocs), but that just asks mike for whatever the default label is and again mike isn't there.  In other words, it just replaces {{version}} with {'provider':'mike','default':v3.0.0'} which still has to be negotiated.

What we can do is put in a repoVersion section in mkdocs and then use the repoVersion to translate a placeholder.  This is what open amt toolkit does.  So the mkdocs section would look something like this:

repoVersion:
   version: 'v3.0.0'
   
The actual string next to 'version' must match the repository tag.

The nice part of this is that we could have many of these tags depending on the repository (example: 'version', 'examples-version', 'compose-version', ...}.  This would allow us to use different versions for different repositories if we so choose.

The bad part of this is that all {{version}} references in the pages would need to be switched to {{repoVersion:version}} and there are 177 of these.

The other option is we drop mike and the idea of having the version drop down at the top and go back to version script on the left hand menu.

Thoughts and preference?

Reply:
Looks like we have a name clash with the name `version`. We should be able to simply change the macro name [`version`](https://github.com/edgexfoundry/edgex-docs/blob/main/template_macros.yaml#L11)  to `edgex-version` and do a search and replace all `{{version}}` with `{{edgex-version}}`.

Reply:
OK - @lenny-intel , sounds like your preference is to still go with mike and make the swap out in all docs ({{version}} to something - perhaps edgex-version).  
@ernestojeda - you ok with that?
I'll cover this tomorrow in TSC.

Reply:
@lenny-intel - I ran into troubles using edgex-version.  It tried to parse on the dash.  edgex.version will work or edgexversion works for me.  If you tried it with the dash and got it to work, pass me your knowledge.  Otherwise, I am submitting a PR with edgexversion and changes to the 177 pages.

Reply:
Ack...can't do a separate PR because it takes latest mkdocs, removal of javascript for version select, addition of mike, etc.  Just going to have to submit as one big PR.